keps/sig-testing/3041-node-conformance-and-features/README.md

```diff
@@ -0,0 +1,782 @@
+capabilities. This flag will not indicate the Feature Gate state.
+
+Proposed change:
+
+> * `[Feature:.+]`: If a test validates functionality that may only work when
+>   specific node capabilities are enabled or when test depends on the
+>   functionality of underlying components like container runtime, and thus may
+>   need to be skipped on certain environments, it receives a `[Feature:.+]`
+>   label, e.g. `[Feature:AppArmor]`. This label can also be applied to signify that
+>   test validates non-core functionality, like `[Feature:Ingress]`.
+>   `[Feature:.+]` tests are not run in a core suites, but typically can run together
+>   on the standard environment as many capabilities and functionality of underlying
+>   components are pre-enabled on the standard environment. Use `[Environment:Foo]` to
+>   signify the specific environment configuration needed to run the test.
+>   For example if test requires GPU to be configured, but tests the non-optional
+>   feature like a device plugin.
+>   `[Feature:+]` label should not be confused with `[FeatureGate:]` label.
+
+And another two:
+
+> * `[FeatureGate:.+]`: If a test only works when the certain feature gate is
+>   enabled it receives a `[FeatureGate:.+]` label. `[FeatureGate:.+]` tests
+>   must also be marked with the status of this feature gate: `[Alpha]`,
+>   `[Beta]`, `[Stable]`, `[Deprecated]`. This label helps to skip tests that
+>   should not work on specific k8s distributive that has a certain feature gate
+>   disabled. This label has to be removed when feature gate value is "locked" or
+>   removed.
+
+> * `[Environment:.+]`: If test requires non-standard environment to run it
+>   receives the `[Environment:.+]` tag. Typically only tests with the matching
+>   `[Environment:.+]` tags can run together. Examples may be GPU needs to be
+>   provisioned, Memory Swap enabled, high-memory or high-CPU machines are needed for
+>   Performance environment, etc.
+
+### Feature label clean up
+
+It is already true today that only a handful of tests marked as `Feature` don't
+have `Serial` or `Disruptive` labels as well. Out of all tests that has `Feature`
+label without `Serial` or `Disruptive`, most of them just degraded,
+and don’t actually need the `Feature` label any longer. Going forward, features
+that require a special environment outside of predefined `LinuxOnly`, `Serial`, etc.
+will need to define their "custom" labels and be commented with the description
+of the special environment needed.
+
+### Clean up labels that means whether special environment is needed
+
+#### NodeSpecialFeature
+
+The label `NodeSpecialFeature:` was introduced in [this document](https://docs.google.com/document/d/1BdNVUGtYO6NDx10x_fueRh_DLT-SVdlPC_SsXjYCHOE/edit?usp=sharing),
+but was never consistently used. Today we only have a couple uses of it that
+might be cleaned up in favor of using `Feature:` tag when applicable.
```

Comment:
NodeSpecialFeature is basically a combination Feature + Environment tag as originally defined so 👍